### PR TITLE
feat: update ontology_info.json for 5.1

### PR DESCRIPTION
**Pull request by Bento007**
## Reason for Change

- https://github.com/chanzuckerberg/cellxgene-ontology-guide/issues/80

### CODE REVIEW

**Review by nayib-jose-gloria — COMMENTED** [review]
looks like the new EFO release has an unexpected type for "synonym". we'll have to investigate...

**Comment by Bento007** [comment]
```
  "EFO:0022503": {
    "ancestors": {
      "EFO:0002888": 1,
      "EFO:0001639": 2,
      "EFO:0000001": 6
    },
    "label": "92-1",
    "description": "Human uveal melanoma",
    "synonyms": [
      92.1
    ],
    "deprecated": false
  }
 ```
It looks like something that could easily be a typo. `92.1` could be converted to a string and then it work work fine.

**Comment by brianraymor** [comment]
Not commenting on the typing, but it looks like EFO is adding cell lines? 

[Human uveal melanoma](https://www.ebi.ac.uk/ols4/ontologies/efo/classes/http%253A%252F%252Fwww.ebi.ac.uk%252Fefo%252FEFO_0022503)

**Comment by brianraymor** [comment]
Typed as decimal?

`<oboInOwl:hasExactSynonym rdf:datatype="http://www.w3.org/2001/XMLSchema#decimal">92.1</oboInOwl:hasExactSynonym>`

**Comment by Bento007** [comment]
It's being parsed into a float in python

**Comment by nayib-jose-gloria** [comment]
> ```
>   "EFO:0022503": {
>     "ancestors": {
>       "EFO:0002888": 1,
>       "EFO:0001639": 2,
>       "EFO:0000001": 6
>     },
>     "label": "92-1",
>     "description": "Human uveal melanoma",
>     "synonyms": [
>       92.1
>     ],
>     "deprecated": false
>   }
> ```
> 
> It looks like something that could easily be a typo. `92.1` could be converted to a string and then it work work fine.

I think its probably safe to add a str cast to anything in synonym

File: tools/ontology-builder/src/all_ontology_generator.py [hunk 1]
```diff
@@ -195,9 +195,8 @@ def _extract_ontology_term_metadata(onto: owlready2.entity.ThingClass) -> Dict[s
         if getattr(onto_term, "IAO_0000115", None):
             term_dict[term_id]["description"] = onto_term.IAO_0000115[0]
         # optional synonym list, if available
-        if getattr(onto_term, "hasExactSynonym", None):
-            term_dict[term_id]["synonyms"] = onto_term.hasExactSynonym
-
+        if hasExactSynonym := getattr(onto_term, "hasExactSynonym", None):
+            term_dict[term_id]["synonyms"] = [x if isinstance(x, str) else str(x) for x in hasExactSynonym]
```
Comment (nayib-jose-gloria) [inline tools/ontology-builder/src/all_ontology_generator.py:199]: nit: any reason not to just [str(x) for x in hasExactSynonym?

**Review by nayib-jose-gloria — APPROVED** [review]

**Pull request merged**